_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,157 @@
+
+```
+/message1 + /message2 + "suffix"
+```
+
+## Event tagging
+
+Data Prepper 2.3 supports tagging events while using the `grok` processor. Events can be tagged optionally be tagged using the following configuration:
+
+```
+processor:
+   - grok:
+        match:
+          message: <pattern>
+        tags_on_match_failure: ["grok_match_fail", "another_tag"]
+```
+
+The presence of tags can be checked in conditional expressions in different processors or routing using the `hasTags()` function to perform conditional ingestion. For example, a conditional expression checking for the presence of tag `grok_match_fail` would be `hasTags("grok_match_fail")`.
+
+## Enhancements to `add_entries` processor
+
+The `add_entries` processor has been enhanced to support values with expressions and to set event metadata keys.
+
+### Expression-based value
+
+The `add_entries` processor has been enhanced to support adding values based on an expression where the return type can be Boolean, integer or float, or string. The following example shows an `add_entries` processor configuration with a `value_expression` option:
+
+```
+processor:
+   - add_entries:
+       entries:
+         - key: "request_len"
+           value_expression: "length(/request)"
+```
+
+This configuration adds an entry with the key `request_len` with a value equal to the length of the request field in the key. The value expression can be any of the expressions supported. See [Supported operators](https://github.com/opensearch-project/data-prepper/blob/main/docs/expression_syntax.md) for more details.
+
+### Setting event metadata keys
+
+The `add_entries` processor can add entries in the event’s metadata instead in the event itself. The following example shows an `add_entries` processor configuration for adding an entry to metadata:
+
+```
+processor:
+   - add_entries:
+       entries:
+         - metadata_key: "request_len"
+           value_expression: "length(/request)"
+```
+This configuration adds an attribute to event metadata with the attribute key `request_len` with value equal to the length of the request field in the key. The value can be set using the value, format, or `value_expression` options of the entries field.
+
+
+## Amazon S3 sink
+
+Data Prepper now supports saving data to Amazon Simple Storage Service (Amazon S3) sinks as Newline delimited JSON (ndjson). Amazon S3 is a popular choice for storing large volumes of data reliably and cost effectively.
+
+Ingesting data into Amazon S3 offers many possibilities for your data pipelines, including some of the following:
+
+* Noisy or uninteresting data can be routed into Amazon S3 and not to OpenSearch in order to reduce load on your OpenSearch cluster. This can help you save on compute and storage costs.
+* Ingestion data into Amazon S3 to have normalized data on-hand that you can use for future processing.
```

Comment:
Please clarify the second bullet. For example, "Ingesting data into Amazon S3 normalizes the data you have on-hand for future processing."